src/pybind/mgr/dashboard/frontend/src/app/shared/components/delete-confirmation-modal/delete-confirmation-modal.component.ts

```diff
@@ -29,7 +29,10 @@ export class DeleteConfirmationModalComponent extends BaseModal implements OnIni
     @Optional() @Inject('submitAction') public submitAction?: Function,
     @Optional() @Inject('backAction') public backAction?: Function,
     @Optional() @Inject('bodyTemplate') public bodyTemplate?: TemplateRef<any>,
-    @Optional() @Inject('bodyContext') public bodyContext?: object,
+    @Optional() @Inject('subHeading') public subHeading?: string,
+    @Optional()
+    @Inject('bodyContext')
+    public bodyContext?: { warningMessage?: string; [key: string]: any },
```

Comment:
it'll be probably nice to create an interface for bodyContext and leave it in this PR so that people will know in the future what all values are supported by the bodyContext.